test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/DataCollection/InProcDataCollectionExtensionManagerTests.cs

```diff
@@ -53,6 +53,25 @@ public InProcDataCollectionExtensionManagerTests()
         _inProcDataCollectionManager = new TestableInProcDataCollectionExtensionManager(_settingsXml, _mockTestEventsPublisher.Object, _defaultCodebase, _testPluginCache, _mockFileHelper.Object);
     }
 
+    [TestMethod]
+    public void CheckExtensionPathsDuplicates()
+    {
+        var testPluginCache = new TestPluginCache();
+        testPluginCache.UpdateExtensions(new List<string> { @"c:\test1\Collector.dll" }, false);
+        testPluginCache.UpdateExtensions(new List<string> { @"c:\test1\Collector.dll", @"c:\test2\Collector.dll" }, true);
```

Comment:
If the previous call is required, could you add a small comment in the code that explains why we need these 2 passes?